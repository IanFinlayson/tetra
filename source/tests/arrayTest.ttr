def main():
   val = ["cccc","reghjgh","abbc","aaaaa"]
   print("Just assigned val and val = ", val, "\n")
   val[0] = "dfkjghkjghsdgksgsggjk"
   print("Just set val[0] = 'dfkjghkjghsdgksgsggjk' and val = ", val, "\n")
   val[1] = "zzzzzzzzz"
   print("Just set val[1] = 'zzzzzzzzz' and val = ", val, "\n")
   val[2] = "ddddddddd"
   print("Just set val[2] = 'ddddddddd' and val = ", val, "\n")
   val[3] = "wwwwwwwww"
   print("Just set val[3] = 'wwwwwwwww' and val = ", val, "\n")
   x = val[0]
   print("Just set x = val[0] and x = ", x, "\n")
   if x != val[1]:
      print(" x != val[1]. x = ", x, "and val[1] = ", val[1],"\n")
      val[2] = val[1]
      print("Just set val[2] = val[1] and val[2] = ",val[2], "and val[1] =",val[1], "\n")
      z = 2
      print("Just set z = 2 and z = ", z, "\n")
   else:
      print(" x = val[1]. x = ", x, "and val[1] = ", val[1],"\n")
      val[1] = val[2]
      print("Just set val[1] = val[2] and val[1] = ",val[1], "and val[2] =",val[2], "\n")
      y = 4
      print("Just set y = 4 and y = ", y, "\n")

   for i in val:
      print("In for loop, i = ", i, "\n")
      q = i
      print("Just set q = i and q = ",q, "\n")

   nums = [1,2,3,4]
   print("Just assigned nums and nums = ", nums, "\n")
   sum = 0
   print("Just assigned sum and sum = ", sum, "\n")
   for number in nums:
      print("In for loop, number = ", number, "\n")
      sum += number
      print("Just added number to sum and sum = ", sum, "\n")
