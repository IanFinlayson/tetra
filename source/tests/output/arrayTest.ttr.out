Just assigned val and val = ['cccc', 'reghjgh', 'abbc', 'aaaaa']
Just set val[0] = 'dfkjghkjghsdgksgsggjk' and val = ['dfkjghkjghsdgksgsggjk', 'reghjgh', 'abbc', 'aaaaa']
Just set val[1] = 'zzzzzzzzz' and val = ['dfkjghkjghsdgksgsggjk', 'zzzzzzzzz', 'abbc', 'aaaaa']
Just set val[2] = 'ddddddddd' and val = ['dfkjghkjghsdgksgsggjk', 'zzzzzzzzz', 'ddddddddd', 'aaaaa']
Just set val[3] = 'wwwwwwwww' and val = ['dfkjghkjghsdgksgsggjk', 'zzzzzzzzz', 'ddddddddd', 'wwwwwwwww']
Just set x = val[0] and x = dfkjghkjghsdgksgsggjk
 x != val[1]. x = dfkjghkjghsdgksgsggjkand val[1] = zzzzzzzzz
Just set val[2] = val[1] and val[2] = zzzzzzzzzand val[1] =zzzzzzzzz
Just set z = 2 and z = 2
In for loop, i = dfkjghkjghsdgksgsggjk
Just set q = i and q = dfkjghkjghsdgksgsggjk
In for loop, i = zzzzzzzzz
Just set q = i and q = zzzzzzzzz
In for loop, i = zzzzzzzzz
Just set q = i and q = zzzzzzzzz
In for loop, i = wwwwwwwww
Just set q = i and q = wwwwwwwww
Just assigned nums and nums = [1, 2, 3, 4]
Just assigned sum and sum = 0
In for loop, number = 1
Just added number to sum and sum = 1
In for loop, number = 2
Just added number to sum and sum = 3
In for loop, number = 3
Just added number to sum and sum = 6
In for loop, number = 4
Just added number to sum and sum = 10

